airbyte-integrations/connectors/source-twilio/source_twilio/streams.py

```diff
@@ -0,0 +1,376 @@
+
+    def parse_response(self, response: requests.Response, stream_state: Mapping[str, Any], **kwargs) -> Iterable[Mapping]:
+        """
+        :return an iterable containing each record in the response
+        """
+        records = response.json().get(self.data_field, [])
+        if self.changeable_fields:
+            for record in records:
+                for field in self.changeable_fields:
+                    record.pop(field, None)
+                    yield record
+        yield from records
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], next_page_token: Mapping[str, Any] = None, **kwargs
+    ) -> MutableMapping[str, Any]:
+        params = super().request_params(stream_state=stream_state, next_page_token=next_page_token, **kwargs)
+        params["PageSize"] = self.page_size
+        if next_page_token:
+            params.update(**next_page_token)
+        return params
+
+
+class IncrementalTwilioStream(TwilioStream, ABC):
+    cursor_field = "date_updated"
+    time_filter_template = "%Y-%m-%dT%H:%M:%SZ"
+
+    def __init__(self, start_date: str = None, **kwargs):
+        super().__init__(**kwargs)
+        self._start_date = start_date
+
+    @property
+    @abstractmethod
+    def incremental_filter_field(self) -> str:
+        """
+        return: date filter query parameter name
+        """
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        """
+        Return the latest state by comparing the cursor value in the latest record with the stream's most recent state object
+        and returning an updated state object.
+        """
+        latest_benchmark = pendulum.parse(latest_record[self.cursor_field], strict=False).strftime(self.time_filter_template)
+        if current_stream_state.get(self.cursor_field):
+            return {self.cursor_field: max(latest_benchmark, current_stream_state[self.cursor_field])}
+        return {self.cursor_field: latest_benchmark}
+
+    def request_params(self, stream_state: Mapping[str, Any], **kwargs) -> MutableMapping[str, Any]:
+        params = super().request_params(stream_state=stream_state, **kwargs)
+        start_date = stream_state.get(self.cursor_field) or self._start_date
+        if start_date:
+            params.update({self.incremental_filter_field: pendulum.parse(start_date, strict=False).strftime(self.time_filter_template)})
+        return params
+
+    def read_records(self, stream_state: Mapping[str, Any] = None, **kwargs):
+        stream_state = stream_state or {}
+        records = super().read_records(stream_state=stream_state, **kwargs)
+        for record in records:
+            record[self.cursor_field] = pendulum.parse(record[self.cursor_field], strict=False).strftime(self.time_filter_template)
+            yield record
+
+
+class TwilioNestedStream(TwilioStream):
+    url_base = "https://api.twilio.com"
```

Comment:
I might be missing something, but it seems like all the url bases that use `https://api.twilio.com` also use `https://api.twilio.com/2010-04-01/` so maybe we can skip setting this in the nested stream and classes that inherit from the nested stream. It's somewhat confusing now to track all the different places where this is set.